### PR TITLE
a11y: add ariaLabel props to EditableField and TextFields

### LINKED ISSUE #2140: [Accessibility][Label and Instruction - Property Pane_Ask a question (Bot Asks)]: Ensures every form element has a label
**User Experience** 
Users who rely on AT will face difficulty to locate the focus if the alt attribute is not defined for the image.

**Test Environment**
Browser: Microsoft Edge Canary Version: 82.0.425.0 (Official build)
OS build: 2004 (19564.1005)
URL: [Bot Framework Composer](http://localhost:3000/dialogs/Main)
[**Prerequisite Here**](https://github.com/microsoft/BotFramework-Composer)

**Repro Steps:** 
1. Open the above mentioned URL in EDGE.
2. "Bot Framework Composer" homepage will gets opened.
3. Select any of the previously created bot from "Recent Bots" or create one from the available Bots under 'Examples' and activate it.
4. Navigate to already created dialog tree item and activate it. Such as- Greetings, Actions, EndDialog etc.
5. Click on "+" and select 'Ask a question'. Then select any of the available options except Oauth login.
6. Navigate to property pane appeared at right side of the screen and select 'Bot Asks'.
7. Run Accessibility insight and observe the issue.

**Actual:** 
Bot Framework logo image doesn't have alternate text or a role of none or presentation.

**Expected:**
Ensures Bot Framework logo image have alternate text or a role of none or presentation.

**Note:**
This accessibility issue was found using Accessibility Insights for Web 2.14.1 (axe-core 3.3.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at [http://aka.ms/AccessibilityInsights](http://aka.ms/AccessibilityInsights).

**Observation:**
Same issue observe throughout the website for every property pane.

**Element path:**
.css-vaviek;#TextField942

**Snippet:**
`<input type="text" id="TextField942" placeholder="ActionsSample-0111" autocomplete="off" class="ms-TextField-field field-157" aria-invalid="false" value="ActionsSample-0111">`

**How to fix:**
    Fix any of the following:
      aria-label attribute does not exist or is empty
      aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
      Form element does not have an implicit (wrapped) <label>
      Form element does not have an explicit <label>
      Element has no title attribute or the title attribute is empty

**MAS Impacted:** [MAS3.3.2](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={4b95a70d-258c-474a-820e-5c8b79d5c0bf})

**Attachment:**
Element must have label
Element must have labelother

### PR DESCRIPTION
**Pull request by beyackle**
This adds a new prop to EditableField so we can wire through the accessibility information wherever it's needed.

## Description

EditableField uses a TextField under the hood, which already supports "ariaLabel" as a prop, so we just need to wire that through directly as a new prop and then set sensible labels on all the EditableFields.

## Task Item

Closes #2140

### CODE REVIEW

**Comment by beyackle** [comment]
Note: don't approve this yet; it's just up for collaboration for now.

File: Composer/packages/extensions/obiformeditor/src/Form/ArrayFieldTemplate/ObjectArray.tsx [hunk 1]
```diff
@@ -108,6 +108,7 @@ const ObjectArray: React.FunctionComponent<ArrayFieldTemplateProps> = props => {
                           }
                         : {}),
                     }}
+                    ariaLabel={formatMessage('Object')}
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/ArrayFieldTemplate/ObjectArray.tsx]: We'd probably want this to be the same as the label right?

File: Composer/packages/extensions/obiformeditor/src/Form/ArrayFieldTemplate/StringArray.tsx [hunk 2]
```diff
@@ -54,6 +54,7 @@ const StringArray: React.FunctionComponent<ArrayFieldTemplateProps> = props => {
             }}
             styles={{ root: { width: '100%' } }}
             data-testid="string-array-text-input"
+            ariaLabel={formatMessage('Value')}
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/ArrayFieldTemplate/StringArray.tsx]: We might want to say something like "New Item: Press enter to add"

File: Composer/packages/extensions/obiformeditor/src/Form/fields/EditableField.tsx [hunk 3]
```diff
@@ -45,6 +46,8 @@ export const EditableField: React.FC<EditableFieldProps> = props => {
     borderColor = localValue || transparentBorder ? 'transparent' : NeutralColors.gray30;
   }
 
+  console.log(ariaLabel);
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/fields/EditableField.tsx]: Oops. remove this

File: Composer/packages/extensions/obiformeditor/src/Form/widgets/TextWidget.tsx [hunk 4]
```diff
@@ -130,6 +130,10 @@ export function TextWidget(props: ITextWidgetProps) {
             },
           },
         }}
+        ariaLabel={
+          // if we hide the widget label above, ARIA will need the label to exist here instead
+          hideLabel ? label : undefined
+        }
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/widgets/TextWidget.tsx]: :+1:

File: Composer/packages/extensions/obiformeditor/src/Form/fields/EditableField.tsx (hunk 3 above)
Comment (beyackle) [reply Composer/packages/extensions/obiformeditor/src/Form/fields/EditableField.tsx]: ✔

**Comment by corinagum** [comment]
Only thing with making the placeholder same as the label is that sometimes the screenreader will read both. Should check for this before merging :)

**Pull request merged**